README.dev.md

```diff
@@ -45,13 +45,86 @@ git config --local core.hooksPath .githooks
+```bash
+tree-sitter build
+tree-sitter parse /path/to/fortran/file.f90
+```
+5. Map the `Violation` struct to a rule code in `fortitude/src/rules/mod.rs`.
+   - `code_to_rule` is never called directly, but the match statement within is
+     analysed by the macro `fortitude_macros::map_codes` to define a `Rule` enum
+     and many associated utilities.
+   - The first two digits for a rule code normally define a subcategory, while the
+     last digit denotes the specific rule within that subcategory. The last digit
+     should not be zero. For example, `T04x` defines rules related to the use of
+     assumed-size arrays and character strings.  This isn't stringently
+     enforced, but you may be asked to renumber the rule if other developers
+     think it would better fit somewhere else.
+   - New rules should be in `RuleGroup::Preview`.
+6. Add a [test](#rule-testing-fixtures-and-snapshots) for your rule. Try to
+   consider edge cases and any scenarios where false positives could occur.
+7. Update the generated documentation using `cargo dev generate-all`.
+
+For some rules, it may be possible to automatically apply a fix for the user,
+though it isn't essential to include a fix when adding a new rule. These are
+typically applied using
+[`Fix`](https://github.com/astral-sh/ruff/blob/main/crates/ruff_diagnostics/src/fix.rs)
+and
+[`Edit`](https://github.com/astral-sh/ruff/blob/main/crates/ruff_diagnostics/src/edit.rs)
+from Ruff. A fix may be one of:
+
+- 'Safe': Applying the fix is guaranteed to not change the behaviour of the
+  user's program. This will normally only apply to stylistic changes.
+- 'Unsafe': Applying the fix may change the behaviour of the user's program in
+  some edge cases.
+- 'Display only': Fortitude can guess at a solution, but makes no guarantees
+  to its correctness or safety.
+
+If you help writing rules, we recommend checking the implementation of existing
+rules to see if anything similar already exists. You can also raise a draft pull
+request to ask for assistance from other developers.
+
+### Naming and Categorising Rules
+
+Similarly to
+[Ruff](https://docs.astral.sh/ruff/contributing/#rule-naming-convention), the name
+of a rule should describe the pattern the rule is intended to fix. Words such as
+'forbid' should be omitted. For example, the name for the rule that warns of
+overly long lines is `LineTooLong`, and not something like `AvoidLineTooLong` or
+`KeepLinesShort`.
+
+Rules should also be categorised appropriately. For example, if a rule is
+intended to discourage the use of outdated features, it may go under
+`Obsolescent`, or perhaps a more specific category such as `Modules` or
+`Typing`. If the rule only affects code readability, it should go under `Style`.
+
+The boundaries between categories are not always clear, so the exact name and
+category of a rule is often determined following a discussion after a pull
+request has been raised.
+
+#### Rule Testing: Fixtures and Snapshots
```

Comment:
Very minor, but I think this is one level too deep? It looks like it's under `Naming and Categorising Rules`

Reply:
Whoops! Well caught, that was just a typo